libcodechecker/server/api/store_handler.py

```diff
@@ -234,52 +231,42 @@ def addCheckerRun(session, storage_session, command, name, tag, username,
             if run_history:
                 run_history.version_tag = None
                 session.add(run_history)
-                session.flush()
 
         run_history = RunHistory(run_id, tag, username, run_history_time)
         session.add(run_history)
-        session.flush()
 
-        storage_session.start_run_session(run_id, session)
+        session.flush()
         return run_id
     except Exception as ex:
-        session.close()
```

Comment:
What is the reason this line being removed?

Reply:
The session will be closed automatically when the execution leaves the `DBSession` context. There is no need to close it there.

Reply:
What are the commit rules for these? When `DBSession.__exit__`, is it just a `close()` without a commit, right?